### ISSUE TITLE
Extensions not working unless vscode is started from terminal

### ISSUE THREAD

**Issue by gennaro-tedesco** [Microsoft/vscode#54033]
- VSCode Version: version 1.25
- OS Version: macOS Sierra 10.12.6

After the last update (see version above), extensions do not work anymore unless vscode is started from terminal.

Steps to Reproduce:

1. launch code from the applications, not from terminal
2. use any extension (for instance Python, LaTeX Workshop, Excel Viewer)

Does this issue occur when all extensions are disabled?: Yes/No

No, as the issue itself is about extensions working properly.

**Comment by joaomoreno** [comment]
What do you mean by _extensions do not work anymore_. What is the outcome?

**Comment by gennaro-tedesco** [comment]
The outcome depends very much on the type of extension; to give a few examples:

- extensions that are supposed to, say, "format" the code (previews, code indentation, alignment, beautify etc...) do not do as intented: running the corresponding commands results in nothing being done.

- extensions that are supposed to run and/or compile language do not do so (Python, LaTeX, SQL): the corresponding commands disappear from the command palette, so somehow they cannot even be activated.

All the above misbehaviours do not occurr if vscode is launched from terminal, in which case it works just fine as always.

**Comment by joaomoreno** [comment]
Can you run <kbd>F1</kbd>, `Developer: Toggle Developer Tools`, any errors there?

**Comment by gennaro-tedesco** [comment]
I do not see any errors being raised when in turn using the extensions: is there anything in particular I should look for (in the developer tool) when executing some particular command?

**Comment by joaomoreno** [comment]
Mainly the Console tab in the Developer tools... it will print errors if there are any.

**Comment by joaomoreno** [comment]
There must be some environmental change in those two, so I ask you to do the following steps twice, once when launching from the terminal and again launching from the applications:

1. <kbd>F1</kbd>, `Developer: Toggle Developer Tools`
2. Open the `Console` tab
3. Type `process.env`

Show me the full output of `process.env` in those two different scenarios.

**Comment by joaomoreno** [comment]
Unfortunately that doesn't seem to be the **full** output... can you double check that you're giving me valid JSON?

Maybe try with `JSON.stringify(process.env)` and copy the whole string.

**Comment by gennaro-tedesco** [comment]
Let me try again: from terminal:
```
{"iTerm.app","TERM"
"xterm-256color","SHELL"
"/bin/zsh","TMPDIR"
"/var/folders/r7/50n1r47n57jb4_pzz8szr6fj0wm2m7/T/","Apple_PubSub_Socket_Render"
"/private/tmp/com.apple.launchd.PxJfprsHAz/Render","TERM_PROGRAM_VERSION"
"3.1.6","TERM_SESSION_ID"
"w0t0p0
678F8F2A-2577-4487-A1E3-47F63E8994AA","ZSH"
"/Users/<user>/.oh-my-zsh","USER"
"<user>","COMMAND_MODE"
"unix2003","SSH_AUTH_SOCK"
"/private/tmp/com.apple.launchd.SaWSNHGcGs/Listeners","__CF_USER_TEXT_ENCODING"
"0x41CA0A87
0x0
0x2","PAGER"
"less","LSCOLORS"
"Gxfxcxdxbxegedabagacad","PATH"
"/Users/<user>/anaconda/bin
/Users/<user>/shell_scripts
/Library/Frameworks/Python.framework/Versions/3.6/bin
/usr/local/bin
/usr/bin
/bin
/usr/sbin
/sbin
/Library/Frameworks/Mono.framework/Versions/Current/Commands
/usr/local/munki
/Library/TeX/texbin
/Applications/Xamarin Workbooks.app/Contents/SharedSupport/path-bin","LANG"
"en_US.UTF-8","ITERM_PROFILE"
"Default","XPC_FLAGS"
"0x0","XPC_SERVICE_NAME"
"0","SHLVL"
"2","HOME"
"/Users/<user>","COLORFGBG"
"12;8","ITERM_SESSION_ID"
"w0t0p0
678F8F2A-2577-4487-A1E3-47F63E8994AA","LESS"
"-R","LOGNAME"
"<user>","LC_CTYPE"
"en_US.UTF-8","SECURITYSESSIONID"
"186a6","COLORTERM"
"truecolor","_"
"/Applications/Visual Studio Code.app/Contents/MacOS/Electron","VSCODE_CLI"
"1","ELECTRON_NO_ATTACH_CONSOLE"
"1","VSCODE_NLS_CONFIG"
"{\"locale\"
\"en-gb\",\"availableLanguages\"
{}}","VSCODE_NODE_CACHED_DATA_DIR_13829"
"/Users/<user>/Library/Application Support/Code/CachedData/0f080e5267e829de46638128001aeb7ca2d6d50e","VSCODE_LOGS"
"/Users/<user>/Library/Application Support/Code/logs/20180713T142755","VSCODE_IPC_HOOK"
"/Users/<user>/Library/Application Support/Code/1.25.0-main.sock","VSCODE_PID"
"13829"}
```

from Applications, instead
```
{"TMPDIR"
"/var/folders/r7/50n1r47n57jb4_pzz8szr6fj0wm2m7/T/","__CF_USER_TEXT_ENCODING"
"0x41CA0A87
0x0
0x2","SHELL"
"/bin/zsh","HOME"
"/Users/<user>","Apple_PubSub_Socket_Render"
"/private/tmp/com.apple.launchd.PxJfprsHAz/Render","SSH_AUTH_SOCK"
"/private/tmp/com.apple.launchd.SaWSNHGcGs/Listeners","PATH"
"/Users/<user>/anaconda/bin
/Users/<user>/shell_scripts
/Library/Frameworks/Python.framework/Versions/3.6/bin
/usr/local/bin
/usr/bin
/bin
/usr/sbin
/sbin
/Library/Frameworks/Mono.framework/Versions/Current/Commands
/usr/local/munki
/Library/TeX/texbin
/Applications/Xamarin Workbooks.app/Contents/SharedSupport/path-bin","LOGNAME"
"<user>","XPC_SERVICE_NAME"
"com.microsoft.VSCode.18744","USER"
"<user>","XPC_FLAGS"
"0x0","VSCODE_NLS_CONFIG"
"{\"locale\"
\"en-gb\",\"availableLanguages\"
{}}","VSCODE_NODE_CACHED_DATA_DIR_38130"
"/Users/<user>/Library/Application Support/Code/CachedData/0f080e5267e829de46638128001aeb7ca2d6d50e","VSCODE_LOGS"
"/Users/<user>/Library/Application Support/Code/logs/20180716T103526","VSCODE_IPC_HOOK"
"/Users/<user>/Library/Application Support/Code/1.25.0-main.sock","VSCODE_PID"
"38130","LANG"
"en_US.UTF-8","LC_CTYPE"
"en_US.UTF-8","LESS"
"-R","LSCOLORS"
"Gxfxcxdxbxegedabagacad","OLDPWD"
"/","PAGER"
"less","PWD"
"/","SHLVL"
"0","ZSH"
"/Users/<user>/.oh-my-zsh","_"
"/Applications/Visual Studio Code.app/Contents/MacOS/Electron"}
```

**Comment by joaomoreno** [comment]
Nope, that's still not JSON... Can you use `JSON.stringify(process.env)` and copy the whole string?

**Comment by gennaro-tedesco** [comment]
This is exactly what I did and the above is all I got. What parts are missing?

**Comment by joaomoreno** [comment]
OK then just run this, it will put the correct contents in the clipboard: `require('electron').clipboard.writeText(JSON.stringify(process.env))`.

**Comment by gennaro-tedesco** [comment]
The output that I get from the clipboard is exactly identical to what I posted above (which as you mentioned is not in proper JSON format, but that is what it is); in any case if the problem lies in (probably) mismatching environmental changes, why does the format of the output matter? Can one not just read it from the above?

**Comment by joaomoreno** [comment]
> The output that I get from the clipboard is exactly identical to what I posted above

I do not believe that. Can you please just paste it?

**Comment by gennaro-tedesco** [comment]
Pasting from the clipboard (vscode from terminal) we have:

```
{"__CF_USER_TEXT_ENCODING":"0x41CA0A87:0x0:0x2","TMPDIR":"/var/folders/r7/50n1r47n57jb4_pzz8szr6fj0wm2m7/T/","HOME":"/Users/<user>","SHELL":"/bin/zsh","SSH_AUTH_SOCK":"/private/tmp/com.apple.launchd.SaWSNHGcGs/Listeners","Apple_PubSub_Socket_Render":"/private/tmp/com.apple.launchd.PxJfprsHAz/Render","PATH":"/Users/<user>/anaconda/bin:/Users/<user>/shell_scripts:/Library/Frameworks/Python.framework/Versions/3.6/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/usr/local/munki:/Library/TeX/texbin:/Applications/Xamarin Workbooks.app/Contents/SharedSupport/path-bin","LOGNAME":"<user>","XPC_SERVICE_NAME":"com.apple.xpc.launchd.oneshot.0x10000020.Electron","COMMAND_MODE":"unix2003","USER":"<user>","XPC_FLAGS":"0x0","VSCODE_NLS_CONFIG":"{\"locale\":\"en-gb\",\"availableLanguages\":{}}","VSCODE_NODE_CACHED_DATA_DIR_40499":"/Users/<user>/Library/Application Support/Code/CachedData/1dfc5e557209371715f655691b1235b6b26a06be","VSCODE_LOGS":"/Users/<user>/Library/Application Support/Code/logs/20180716T112218","VSCODE_IPC_HOOK":"/Users/<user>/Library/Application Support/Code/1.25.1-main.sock","VSCODE_PID":"40499","LANG":"en_US.UTF-8","LC_CTYPE":"en_US.UTF-8","LESS":"-R","LSCOLORS":"Gxfxcxdxbxegedabagacad","OLDPWD":"/","PAGER":"less","PWD":"/","SHLVL":"0","ZSH":"/Users/<user>/.oh-my-zsh","_":"/Applications/Visual Studio Code.app/Contents/MacOS/Electron"}
```
where I just replaced my username with `<user>` as I am not on my computer.

From the applications, instead
```
{"__CF_USER_TEXT_ENCODING":"0x41CA0A87:0x0:0x2","TMPDIR":"/var/folders/r7/50n1r47n57jb4_pzz8szr6fj0wm2m7/T/","HOME":"/Users/<user>","SHELL":"/bin/zsh","SSH_AUTH_SOCK":"/private/tmp/com.apple.launchd.SaWSNHGcGs/Listeners","Apple_PubSub_Socket_Render":"/private/tmp/com.apple.launchd.PxJfprsHAz/Render","PATH":"/Users/<user>/anaconda/bin:/Users/<user>/shell_scripts:/Library/Frameworks/Python.framework/Versions/3.6/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/usr/local/munki:/Library/TeX/texbin:/Applications/Xamarin Workbooks.app/Contents/SharedSupport/path-bin","LOGNAME":"<user>","XPC_SERVICE_NAME":"com.apple.xpc.launchd.oneshot.0x10000020.Electron","COMMAND_MODE":"unix2003","USER":"<user>","XPC_FLAGS":"0x0","VSCODE_NLS_CONFIG":"{\"locale\":\"en-gb\",\"availableLanguages\":{}}","VSCODE_NODE_CACHED_DATA_DIR_40499":"/Users/<user>/Library/Application Support/Code/CachedData/1dfc5e557209371715f655691b1235b6b26a06be","VSCODE_LOGS":"/Users/<user>/Library/Application Support/Code/logs/20180716T112218","VSCODE_IPC_HOOK":"/Users/<user>/Library/Application Support/Code/1.25.1-main.sock","VSCODE_PID":"40499","LANG":"en_US.UTF-8","LC_CTYPE":"en_US.UTF-8","LESS":"-R","LSCOLORS":"Gxfxcxdxbxegedabagacad","OLDPWD":"/","PAGER":"less","PWD":"/","SHLVL":"0","ZSH":"/Users/<user>/.oh-my-zsh","_":"/Applications/Visual Studio Code.app/Contents/MacOS/Electron"}
```

**Comment by joaomoreno** [comment]
I think you pasted the same output for both... :confused:

**Comment by gennaro-tedesco** [comment]
sorry, my bad. I amended the comment above.

**Comment by joaomoreno** [comment]
You have edited the second snippet and broke it...

I'm going to ask you one more time. Just paste in here the full output of that command. If you fail to provide the output once more, I'll just close the issue.

**Comment by joaomoreno** [comment]
I'm sorry we couldn't figure out the issue. If you manage to fix it, please let us know.
